File: backend/src/main/java/github/backend/model/Project.java
package github.backend.model;

import jakarta.persistence.*;
import lombok.Data;

import java.util.Set;
@Entity
@Data
@Table(name = "project")
public class Project {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    private String title;

    @Column(length = 1024)
    private String description;

    @ManyToMany(cascade = CascadeType.PERSIST)
    @JoinTable(name = "project_tags",
            joinColumns = @JoinColumn(name = "project_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_id"))
    private Set<Tag> tags;

    @Column(name = "video_url")
    private String videoUrl;
    @Column(name = "github_url")
    private String githubUrl;

}

File: frontend/lib/models/project.dart
class Project {
  final String title;
  final String description;
  final List<String> tags;
  final String? videoUrl;
  final String? githubUrl;

  Project({
    required this.title,
    required this.description,
    required this.tags,
    this.videoUrl,
    this.githubUrl,
  });

  factory Project.fromJson(Map<String, dynamic> json) {
    return Project(
      title: json['title'],
      description: json['description'],
      tags: (json['tags'] as List).map((tag) => tag['name'] as String).toList(),
      videoUrl: json['videoUrl'],
      githubUrl: json['githubUrl'],
    );
  }
}
